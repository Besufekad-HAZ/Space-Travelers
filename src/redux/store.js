// import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
} from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';
import missionsSlice from './missions/slice-Missions';

const reducers = combineReducers({
  rockets: rocketsSlice,
  missions: missionsSlice,
});

const store = createStore(reducers, applyMiddleware(thunk));
export default store;
