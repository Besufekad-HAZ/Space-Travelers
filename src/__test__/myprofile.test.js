import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MyProfile from '../component/myprofile';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('MyProfile component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <MyProfile />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
