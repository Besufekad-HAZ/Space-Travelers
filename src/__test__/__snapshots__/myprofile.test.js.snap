// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component renders correctly 1`] = `
<p
  className="myprofile"
>
  No rockets reserved and No Missions Joined!
</p>
`;
